### DISCUSSION TITLE
Panel library interactive widget doesn't work with jupyterlab and eg

### DISCUSSION THREAD

**Discussion by vlad-pesternikov-unskript** [jupyter/enterprise_gateway#1289] [Q&A]
The chart created by the tutorial is interactive in a vanilla Jupyter instance, but it is NOT interactive in Jupyter with enterprise gateway. This is code of notebook cell:
```
import pandas as pd; import numpy as np; import matplotlib.pyplot as plt

csv_file = 'https://raw.githubusercontent.com/holoviz/panel/main/examples/assets/occupancy.csv'
data = pd.read_csv(csv_file, parse_dates=['date'], index_col='date')

data.tail()


import matplotlib as mpl
mpl.use('agg')

from matplotlib.figure import Figure

def mpl_plot(avg, highlight):
    fig = Figure()
    ax = fig.add_subplot()
    avg.plot(ax=ax)
    if len(highlight): highlight.plot(style='o', ax=ax)
    return fig

def find_outliers(variable='Temperature', window=30, sigma=10, view_fn=mpl_plot):
    avg = data[variable].rolling(window=window).mean()
    residual = data[variable] - avg
    std = residual.rolling(window=window).std()
    outliers = (np.abs(residual) > std * sigma)
    return view_fn(avg, avg[outliers])

find_outliers(variable='Temperature', window=20, sigma=10)


import panel as pn
pn.extension()

window = pn.widgets.IntSlider(name='window', value=50, start=1, end=60)
sigma = pn.widgets.IntSlider(name='sigma', value=10, start=0, end=20)

interactive = pn.bind(find_outliers, window=window, sigma=sigma)

first_app = pn.Column(window, sigma, interactive)

first_app
```

Used this manual: [Panel manual](https://panel.holoviz.org/getting_started/build_app.html)

Could you give any pointer how to troubleshoot this issue?

**Comment by kevin-bates** [comment]
Hi @vlad-pesternikov-unskript - thanks for opening this discussion.  Frankly speaking, I know nothing about how Jupyter widgets work but still have some questions that might trigger some thoughts.

1. Where is the kernel running relative to the EG and "vanilla" configurations?  (For "vanilla" it will be local to the jupyter server.)
2. What are the symptoms that things are not working (besides non-interactivity)?  Are any messages evident in the EG log/console?
3. Are there specific packages required where the kernel is running that perhaps are not installed in the EG configuration?
4. Have you checked the kernel's logging output for messages? (This will be a function of the configuration and process proxy in use - Kubernetes, Hadoop YARN, SSH, Docker.)

I don't know if "widgets" make an assumption that the kernel is local to the browser-facing server or if they require a communication channel other than the websocket that is communicating with the kernel over ZMQ.

**Comment by amit-chandak-unskript** [comment]
Thanks @kevin-bates for your prompt reply. On further debugging, we found the issue to be a missing pip package on the jupyter server side, which is pyviz-comms. Once we install this, it started working with EG as well.

**Reply by kevin-bates** [reply]
Excellent news @amit-chandak-unskript - thank you for the update!
